### PR TITLE
Add Markdown link check to GitHub Actions

### PR DESCRIPTION
**Pull request by brandones**
Adds a GitHub Action that uses https://github.com/gaurav-nelson/github-action-markdown-link-check to check that all the links in our documentation are working. cc @gracepotma

### CODE REVIEW

**Comment by brandones** [comment]
I think this check made an amazing case for itself: https://github.com/openmrs/openmrs-esm-core/runs/2842350437?check_suite_focus=true

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Check documentation
```
Comment (brandones) [inline .github/workflows/docs.yml:1]: This file is the actual change this PR is about; all the rest is just fixing errors that this check identified.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  schedule:
+    - cron: "0 9 * * *"
```
Comment (brandones) [inline .github/workflows/docs.yml:9]: Runs this check daily at 9am, in case any of the external links get stale.

**Review by jonathandick — APPROVED** [review]
LGTM

File: .github/workflows/docs.yml (hunk 2 above)
Comment (FlorianRappl) [reply .github/workflows/docs.yml:9]: Good thinking!

**Review by FlorianRappl — APPROVED** [review]
:beers:

**Pull request merged**